### ISSUE TITLE
Update outdated use of GatsbyReporter

### ISSUE THREAD

**Issue by prichey** [algolia/gatsby-plugin-algolia#97]
Currently the plugin does this ([here](https://github.com/algolia/gatsby-plugin-algolia/blob/master/gatsby-node.js#L3)):

```
const report = require('gatsby-cli/lib/reporter');
```

which doesn't work in newer versions of Gatsby so the logger falls back to `console.log` ([here](https://github.com/algolia/gatsby-plugin-algolia/blob/master/gatsby-node.js#L294)).

The correct approach in Gatsby is to use the [GatsbyReporter](https://www.gatsbyjs.com/docs/node-api-helpers/#GatsbyReporter) which is available to `onPostBuild`, via:

```
exports.onPostBuild = async function ({ graphql, reporter }, options) {}
```

**Comment by prichey** [comment]
I might be able to work on a PR for this after I merge a few other fixes but if anyone else feels like taking this on, it shouldn't be that bad.

**Comment by Haroenv** [comment]
ah thank you, this probably fixes #55, #89 and a few more!
